testsuite/httpx/__init__.py

```diff
@@ -65,6 +65,15 @@ def __str__(self):
         return f"Result[error={self.error}]"
 
 
+class ResultList(list):
+    """List-like object for Result"""
+
+    def assert_all(self, status_code=None):
```

Comment:
```suggestion
    def assert_all(self, status_code):
```

Reply:
agree, we can add named parameters if/when more asserts is needed.